### ISSUE TITLE
TiKV resolved ts gap keep increasing

### ISSUE THREAD

**Issue by fubinzh** [tikv/tikv#14831]
### What version of TiKV are you using?
```
TiKV
Release Version:   7.1.0
Edition:           Community
Git Commit Hash:   76d524f7acb4215ee68f07307e0312b9f56ec866
Git Commit Branch: heads/refs/tags/v7.1.0
UTC Build Time:    2023-05-23 11:47:10
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

```

### What operating system and CPU are you using?

K8s, 16c 32G

### Steps to reproduce
1. CDC changefeed running to replicate data to kafka. and TiDB tidb_enable_row_level_checksum is on
2. There are 2 workload running. workload 2 is insert only, workload 1 with insert/update/select.
3. some chaos testing executed on the cluster like pod failure, network partition, etc.

### What did you expect?
Resolved ts gap should not keep increasing.

### What did happened?
TiKV resolved ts gap keep increasing since ~5/24 19:34
![image](https://github.com/tikv/tikv/assets/7403864/80726f95-9522-4977-9145-e5f5a875a957)

**Comment by tonyxuqqi** [comment]
Is it by design if there's no failure? @fubinzh

**Comment by fubinzh** [comment]
@tonyxuqqi Is it possible to resolve the lock automatically somehow? Though we can unlock it manually following https://github.com/pingcap/tiflow/issues/4516#issuecomment-1049540773, it is quite difficult for a user to itendify/work around such issue.

**Comment by zhangjinpeng1987** [comment]
@fubinzh Does the chaos test always running or just running for a while(10min or 20min) and then stop chaos test let the system to recover by itself?

**Comment by fubinzh** [comment]
> @fubinzh Does the chaos test always running or just running for a while(10min or 20min) and then stop chaos test let the system to recover by itself? I think the first one is a more common case and meaningful for users. cc @tonyxuqqi

Yes, the chaos only lasted for 10m, and we wait for cdc checkpoiny recovery before next chaos injection.  I updated the issue description as well.

**Comment by fubinzh** [comment]
> The TiDB is killed during the test and therefore remaining locks are uncleared until the GC kicks in. The question is if GC is enabled or tikv_gc_life_time is changed during this test. @fubinzh
@tonyxuqqi we didn't change gc configurations during testing explicitely.

**Comment by tonyxuqqi** [comment]
OK, So it's because TiCDC would block GC if TiCDC cannot move forward. But it should still clear locks. Need to figure out why it's not behaved as expected. @nongfushanquan

**Comment by fubinzh** [comment]
Run scenario ha-tidb-random-kill -> recovery -> ha-tikv-random-to-tidb-random-network-partition-last-for-10m for several times today with 7.1.0, on a tidb cluster with 3 pd/tidb/tikv running tpcc workload (100 warehouse, qps ~25k), the issue is not reproduced.
